CMakePresets.json

```diff
@@ -0,0 +1,28 @@
+{
```

Comment:
This file is one generated by CMake and not a part of vscode's settings. I'm just concerned that other IDE will pick this up and attempt to use it, particularly if they aren't on a mac.

Reply:
One reason why I am so eager about this is that such a file allows us to make the building in CI much easier and also provide "suggested" configurations for e.g. headless build which are currently stored in a README - and it would also help me in setting up the wasm build chain.

An IDE may pick this file up, but as long as one doesn't select any preset, Cmake will not start configuring. And even if one applies this by incident, it is always possible to manually configure via the CLI. In general this allows to ditch the CLI completely for 95% of the dev work.

The idea is to support more configurations over time.

Reply:
Ah, actually, this is really cool, changes the way people build, and something I didn't know about!
CLion picks these up and gives them as options.

Some thought...

Could this be its own PR, seems like vscode is actually secondary?

Could we use conditions to disable things...
```
{
    "inherits": "macOS base",
    "name": "macOS_supernova",
    "displayName": "macOS with supernova",
    "cacheVariables": {
        "SUPERNOVA": "ON"
    },
    "condition": {
        "type": "equals",
        "lhs": "${hostSystemName}",
        "rhs": "Darwin"
    }
},
```

Why is this disabling supernova? Having a config for every combination of settings will be too much. Perhaps just two per OS, recommended and testing? 

Should we also use build presets?

Can we use test presets with the sclang unit tests?

Reply:
Actually, scrap that... I think this should be its own PR, and would be a mistake to hide this quite important useful feature in a PR about IDE settings. I'm also happy to contribute and add linux stuff — I know nothing about windows.

Reply:
I think there are multiple ways of writing said file, but other people have to write them for their respective OSes - I think one there are shared functionality there can be talks about how this could be implemented in the best possible manner.
But one thing that I learned from the CI though: don't rely too much on conditions, they create difficult to maintain implicit states, where I would favor code duplication in favor of readability.

I don't understand how you could use this for testing, but I want to actually use this to make the CI setup a bit more manageable.
I think we don't need to limit it to 2 per platform because it is possible to list all presets and add a description for them.

I agree this would have been better to have its own PR, I would have thought that this would be less controversial actually :D I included it b/c it is the preferred way on how to configure C++ projects in vscode.

> Why is this disabling supernova?

I think the default way of building SC is w/o supernova? I always consider it optional, but it would be good to promote it further.

Reply:
I also think that CMakePresets should be a separate PR, maybe preceding the one specifically for VS Code.

> I think the default way of building SC is w/o supernova?

Yes and no :)
Technically the default in CMake is without, but we have included it in the release builds for a while.
One reason for having it as off by default was that it needs additional libraries on macOS (portaudio and fftw). There were also some issues with it running on Windows, so I'm not sure if it's used much on that platform.
I'd personally be in favor of tuning it on by default.